API/models/model.go

```diff
@@ -124,6 +124,29 @@ func PropagateParentIdChange(ctx context.Context, oldParentId, newId string, ent
 	return nil
 }
 
+// PropagateDomainChange: search for all objects with reference to the modified domain
+func PropagateDomainChange(ctx context.Context, oldDomainId, newDomainId string) error {
```

Comment:
I think this function could be in the repository package, since it has a database logic and not a business logic.

Reply:
I kept it there because PropagateParentIdChange is also there, but we can move both to repository then

Reply:
Actually, propagate functions are quite different from the one in repository and more similar to other functions from model accessing the database

Reply:
Moved both propagate to repository, let's slowly fix the partition between model / repository.